client/logging/src/lib.rs

```diff
@@ -0,0 +1,357 @@
+				.parse()
+				.expect("provided directive is valid"),
+		)
+		// Set warn logging by default for some modules.
+		.add_directive(
+			"cranelift_wasm=warn"
+				.parse()
+				.expect("provided directive is valid"),
+		)
+		.add_directive("hyper=warn".parse().expect("provided directive is valid"))
+		// Enable info for others.
+		.add_directive(tracing_subscriber::filter::LevelFilter::INFO.into());
+
+	if let Ok(lvl) = std::env::var("RUST_LOG") {
+		if lvl != "" {
+			// We're not sure if log or tracing is available at this moment, so silently ignore the
+			// parse error.
+			for directive in parse_directives(lvl) {
+				env_filter = env_filter.add_directive(directive);
+			}
+		}
+	}
+
+	for directive in extra_directives {
+		// We're not sure if log or tracing is available at this moment, so silently ignore the
+		// parse error.
+		env_filter = env_filter.add_directive(directive);
+	}
+
+	// If we're only logging `INFO` entries then we'll use a simplified logging format.
+	let simple = match Layer::<FmtSubscriber>::max_level_hint(&env_filter) {
+		Some(level) if level <= tracing_subscriber::filter::LevelFilter::INFO => true,
+		_ => false,
+	};
+
+	// Always log the special target `sc_tracing`, overrides global level.
+	// NOTE: this must be done after we check the `max_level_hint` otherwise
+	// it is always raised to `TRACE`.
+	env_filter = env_filter.add_directive(
+		"sc_tracing=trace"
+			.parse()
+			.expect("provided directive is valid"),
+	);
+
+	let enable_color = atty::is(atty::Stream::Stderr);
+	let timer = ChronoLocal::with_format(if simple {
+		"%Y-%m-%d %H:%M:%S".to_string()
+	} else {
+		"%Y-%m-%d %H:%M:%S%.3f".to_string()
+	});
+
+	let telemetries = if let Some(telemetry_external_transport) = telemetry_external_transport {
+		sc_telemetry::Telemetries::with_wasm_external_transport(telemetry_external_transport)
+	} else {
+		Default::default()
+	};
+	let senders = telemetries.senders();
+	let telemetry_layer = sc_telemetry::TelemetryLayer::new(senders);
+	let event_format = EventFormat {
+		timer,
+		display_target: !simple,
+		display_level: !simple,
+		display_thread_name: !simple,
+		enable_color,
+	};
+	let builder = FmtSubscriber::builder()
+		.with_env_filter(env_filter)
+		.with_writer(
+			#[cfg(not(target_os = "unknown"))]
+			std::io::stderr,
+			#[cfg(target_os = "unknown")]
+			std::io::sink,
+		);
+
+	#[cfg(not(target_os = "unknown"))]
+	let builder = builder.event_format(event_format);
+
+	let subscriber = builder.finish().with(PrefixLayer).with(telemetry_layer);
+
+	#[cfg(target_os = "unknown")]
+	let subscriber = subscriber.with(ConsoleLogLayer::new(event_format));
+
+	Ok((subscriber, telemetries))
+}
+
+// Transform a string of comma separated logging directive into a `Vec<Directive>`.
+fn parse_directives(dirs: impl AsRef<str>) -> Vec<Directive> {
+	let dirs = dirs.as_ref();
+
+	if dirs.is_empty() {
+		return Default::default();
+	}
+
+	dirs.split(',').filter_map(|s| s.parse().ok()).collect()
```

Comment:
```suggestion
	dirs.as_ref().split(',').filter_map(|s| s.parse().ok()).collect()
```

Should do the same in the end?

Reply:
It's not my code :sweat: I have to check I'm not sure.

Reply:
You are correct